aptos-move/framework/aptos-stdlib/sources/math128.spec.move

```diff
@@ -0,0 +1,36 @@
+spec aptos_std::math128 {
+
+    spec max(a: u128, b: u128): u128 {
+        aborts_if false;
+        ensures a >= b ==> result == a;
+        ensures a < b ==> result == b;
+    }
+
+    spec min(a: u128, b: u128): u128 {
+        aborts_if false;
+        ensures a < b ==> result == a;
+        ensures a >= b ==> result == b;
+    }
+
+    spec average(a: u128, b: u128): u128 {
+        pragma opaque;
+        aborts_if false;
+        ensures result == (a + b) / 2;
+    }
+
+    spec pow(n: u128, e: u128): u128 {
+        pragma opaque;
+        // TODO: specify the abort-condition
```

Comment:
Are these dependent on missing features in the Prover? If not, why not do them in the same PR.

Reply:
Lets add the abstract specification (via recursive function) as we discussed before. We do not need to verify the implementation itself.

Reply:
That's a good point. We can specify it as `aborts_if [abstract] spec_pow(n, e) > MAX_U128;`. I previously was not sure because it may not be true in some implementations where there is an unnecessary overflow when `n` and `e` are very large. I think our implementation doesn't have that issue, and we can safely assume the aborts-if spec.